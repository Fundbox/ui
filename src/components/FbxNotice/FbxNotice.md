# FbxNotice
Display centered content on a blue strip.
