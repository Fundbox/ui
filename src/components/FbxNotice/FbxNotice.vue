<template>
  <div class="notice">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'FbxNotice',
}
</script>

<style lang="scss" scoped>
  @import "../../styles/utils/color-palette.scss";

  .notice {
    background-color: $light-blue;
    display: flex;
    align-items: center;
    justify-content: center;
  }
</style>
