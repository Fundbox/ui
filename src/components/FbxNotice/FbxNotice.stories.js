import { storiesOf } from '@storybook/vue'
import { withInfo } from 'storybook-addon-vue-info'

import FbxNotice from './FbxNotice.vue'
import summary from './FbxNotice.md'

const stories = storiesOf('Components/Notice', module)

stories.add('default', withInfo({ summary })(() => ({
  components: { FbxNotice },
  template: `<fbx-notice><div style="padding: 16px">lorem ipsum</div></fbx-notice>`
})))
