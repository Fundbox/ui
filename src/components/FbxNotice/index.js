import FbxNotice from './FbxNotice.vue'
import { registerComponents, vueUse } from '../../utils/plugins'

const components = {
  FbxNotice
}

const VuePlugin = {
  install(Vue) {
    registerComponents(Vue, components)
  }
}

vueUse(VuePlugin)

export default VuePlugin
