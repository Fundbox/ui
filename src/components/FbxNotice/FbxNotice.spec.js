import { shallowMount } from '@vue/test-utils'
import FbxNotice from './FbxNotice.vue'

describe('Components/FbxNotice', () => {
  describe('snapshots', () => {
    it('renders the default correctly', () => {
      const wrapper = shallowMount(FbxNotice, {
        // mounting options https://vue-test-utils.vuejs.org/api/options.html
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
