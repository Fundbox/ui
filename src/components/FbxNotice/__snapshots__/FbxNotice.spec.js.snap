// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FbxNotice snapshots renders the default correctly 1`] = `<div class="notice"></div>`;
